WcaOnRails/app/webpacker/components/Media/MediaTable.jsx

```diff
@@ -0,0 +1,257 @@
+    key: country.iso2,
+    text: country.name,
+    value: country.iso2,
+  }));
+  const continentOptions = continents.map((continent) => ({
+    key: continent.id,
+    text: continent.name,
+    value: continent.name,
+  }));
+  const yearOptions = [
+    {
+      key: 'all',
+      text: 'All Years',
+      value: 'All Years',
+    },
+    ...years.map((year) => ({
+      key: year,
+      text: year,
+      value: year,
+    })),
+  ];
+
+
+  const handleUpdateSuccess = () => {
+    setModalOpen(true);
+  };
+
+  const handleCloseModal = () => {
+    setModalOpen(false);
+  };
+  const [selectedYear, setSelectedYear] = useState(yearOptions[1].value);
+  const [selectedRegion, setSelectedRegion] = useState("All Regions");
+  const { save, saving } = useSaveAction();
+  const [mediaCombined, setMediaCombined] = useState([])
+  const type = isValidate ? 'pending' : 'accepted'
+  const { loading, error, data, sync } = useLoadedData(
+    apiV0Urls.media.list(type, selectedYear, selectedRegion)
+  );
+  const [modalOpen, setModalOpen] = useState(false);
+
+  useEffect(() => {
+    if (data) {
+      setMediaCombined(data?.map((medium, idx) => {
+        const iso2 = medium.competition.country_iso2;
+        const country = countries.byIso2[iso2];
+        return {
+          timestampSubmitted: new Date(medium.timestampSubmitted).toString(),
+          timestampDecided: medium.timestampDecided,
+          id: medium.competition.id,
+          name: medium.competition.name,
+          type: medium.type,
+          text: medium.text,
+          uri: medium.uri,
+          country_iso2: iso2,
+          country_name: country ? country.name : '',
+          city: medium.competition.city,
+          startDate: medium.competition.start_date,
+          endDate: medium.competition.end_date,
+          mediaId: medium.id,
+          competetionDate: dateRangeBetween(medium.competition.start_date, medium.competition.end_date),
+          competetionUrl: competitionUrl(medium.competition.id)
+        };
+      }))
+    }
+  }, [data]);
+  const [modalParams, setModalParams] = React.useState({
+    open: false,
+    mediaId: null,
+    status: null,
+  });
+  const confimMedia = () => {
+    save(apiV0Urls.media.update(modalParams.mediaId), { status: modalParams.status }, () => {
+      sync(); handleUpdateSuccess(); setModalParams({ ...modalParams, open: false })
+    }, { method: 'PATCH' });
+  };
+  const [sortColumn, setSortColumn] = useState(null);
+  const [sortDirection, setSortDirection] = useState(null);
+  const handleSort = (clickedColumn) => () => {
+    if (sortColumn !== clickedColumn) {
+      setSortColumn(clickedColumn);
+      setSortDirection('ascending');
+      setMediaCombined([...mediaCombined].sort((a, b) => a[clickedColumn] - b[clickedColumn]));
+    } else {
+      setMediaCombined([...mediaCombined].reverse());
+      setSortDirection(sortDirection === 'ascending' ? 'descending' : 'ascending');
+    }
+  };
```

Comment:
This could be encapsulated via `useReducer`:
```
const [{ sortColumn, sortDirection }, handleSort] = useReducer(sortReducer, { sortColumn: null, sortDirection: null })
```
(with a definition of `sortReducer` outside the component).

Reply:
Done